locals.tf

```diff
@@ -1,29 +1,17 @@
-# Copyright 2017, 2021 Oracle Corporation and/or affiliates.
+# Copyright 2017, 2023 Oracle Corporation and/or affiliates.
 # Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl
 
 locals {
-  tenancy_id     = coalesce(var.tenancy_id, var.tenancy_ocid)
-  compartment_id = coalesce(
-    var.compartment_id, var.compartment_ocid,
-    var.tenancy_id, var.tenancy_ocid,
-  )
-  user_id = var.user_id != "" ? var.user_id : var.current_user_ocid
-
-  api_private_key = (
-    var.api_private_key != ""
-    ? try(base64decode(var.api_private_key), var.api_private_key)
-    : var.api_private_key_path != ""
-      ? file(var.api_private_key_path)
-      : null)
-
-  bastion_public_ip                      = var.create_bastion_host == true ? module.bastion[0].bastion_public_ip : var.bastion_public_ip != "" ? var.bastion_public_ip: ""
-  operator_private_ip                    = var.create_operator == true ? module.operator[0].operator_private_ip : var.operator_private_ip !="" ? var.operator_private_ip: ""
+  bastion_public_ip                      = var.create_bastion_host == true ? module.bastion[0].bastion_public_ip : var.bastion_public_ip != "" ? var.bastion_public_ip : ""
+  operator_private_ip                    = var.create_operator == true ? module.operator[0].operator_private_ip : var.operator_private_ip != "" ? var.operator_private_ip : ""
   operator_instance_principal_group_name = var.create_operator == true ? module.operator[0].operator_instance_principal_group_name : ""
```

Comment:
There's a condition missing here. This should only be enabled if instance_principal is enabled.

Reply:
Added